### DISCUSSION TITLE
Lighthouse reports bad performance 29/100

### DISCUSSION THREAD

**Discussion by marvinRoellecke** [neuefische/questions#443] [Web] [answered]
Hi,
I run a lighthouse test to check the performance of my app. The result is quite poor -> 29 of 100 points.

I notice that the first page build up takes quite long. After that the app runs as usual.

I did the following steps to locate the root cause with no success 
- use data from local dummy array instead of database fetch
- don't render any images 
- remove complex filter functions (quite long array method chain)

[repository](https://github.com/marvinRoellecke/capstone-project--TypeScript)
[vercel](https://capstone-project-typescript.vercel.app/)

Please see some details of the report below. 
**Maybe anyone has a hint what I can check next.**

Thanks much

<img width="824" alt="Bildschirmfoto 2023-01-09 um 10 14 23" src="https://user-images.githubusercontent.com/115343305/211279853-b36f49b8-dd86-4841-b1e0-cdb2cf350d65.png">

<img width="808" alt="Bildschirmfoto 2023-01-09 um 10 15 04" src="https://user-images.githubusercontent.com/115343305/211279882-c7bdbc96-d8d1-46f9-93c3-790d0d3db6aa.png">

<img width="809" alt="Bildschirmfoto 2023-01-09 um 10 16 41" src="https://user-images.githubusercontent.com/115343305/211279908-f915138f-c388-4245-a86f-a909e71ad494.png">

<img width="794" alt="Bildschirmfoto 2023-01-09 um 10 16 54" src="https://user-images.githubusercontent.com/115343305/211279927-5cff217b-b9df-459f-b76a-0aee7abe2bd9.png">

**Comment by pixelass** [comment, marked as answer]
Please test your app in production (not development on localhost)
Please test in an incognito window WITHOUT any browser extensions. 
You should notice a huge difference.

**Reply by pixelass** [reply]
![Screenshot 2023-01-09 at 11 37 22](https://user-images.githubusercontent.com/1148334/211289306-ee4c6704-cf4d-4d2a-a678-12b70092e8f7.png)

**Reply by marvinRoellecke** [reply]
Running lighthouse on Vercel deployment shows a better performance as the screenshot above shows.

I don't understand the step with the incognito window as a user may open the app in a "normal" window.
At least I nearly get same results running in incognito and "normal" window.

**Reply by pixelass** [reply]
You never know what the user has installed, therefore you can safely ignore any chrome extensions. Lighthouse should not downgrade your report based on YOUR browser but rather try to generalize the results (therefore all extensions should be disabled). Since some extensions still run even when disabled for the current page, it is better to test in a "clean browser"

Google offers a service that does the same on special servers just for lighthouse tests: https://pagespeed.web.dev/report?url=https%3A%2F%2Fcapstone-project-typescript.vercel.app%2F


The tests are provided to ensure that your App has a good performance and score in a "Vanilla browser (clean out of the box, no add-ons installed)"
